sonic_platform_base/watchdog_base.py

```diff
@@ -0,0 +1,46 @@
+#
+# watchdog_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a hardware watchdog module in SONiC
+#
+
+
+class WatchdogBase:
+    """
+    Abstract base class for interfacing with a hardware watchdog module
+    """
+
+    def arm(self, seconds):
+        """
+        Arm the hardware watchdog with a timeout of <seconds> seconds.
+        If the watchdog is currently armed, calling this function will
+        simply reset the timer to the provided value.
+
+        Returns:
+            A boolean, True if watchdog is armed successfully, False if not
+        """
+        raise NotImplementedError
+
+    def disarm(self):
+        """
+        Disarm the hardware watchdog
+
+        Returns:
+            A boolean, True if watchdog is disarmed successfully, False if not
+        """
+        raise NotImplementedError
+
+    def is_armed(self):
+        """
+        Retrieves the armed state of the hardware watchdog and if armed, the
+        number of seconds remaining on the watchdog timer
```

Comment:
If watchdog is implemented by CPLD, say (instead of Host CPU), the number of seconds remaining may not be retrievable/supported.. Can the API be split into 2 : 1 to check if wdog is armed and another to retrieve remaining seconds?

Reply:
Addressed in https://github.com/Azure/sonic-platform-common/pull/13/commits/5fbf599a801665066545d8261ac3e3dfba34d762.

Reply:
Thanks